synapse/handlers/message.py

```diff
@@ -888,6 +888,14 @@ async def create_and_send_nonmember_event(
 
             spam_check = await self.spam_checker.check_event_for_spam(event)
             if spam_check is not synapse.spam_checker_api.Allow.ALLOW:
+                if isinstance(spam_check, tuple):
+                    [code, dict] = spam_check
+                    raise SynapseError(
+                        403,
+                        "This message had been rejected as probable spam",
+                        code,
+                        dict,
+                    )
```

Comment:
This code should be surrounded by a try/except on `ValueError`s so that a buggy module results in an error that's easier to debug.

Reply:
Is this what you had in mind?